tests/vmi_cloudinit_test.go

```diff
@@ -348,13 +351,22 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 
 				By("checking cloudinit network-config")
 				CheckCloudInitFile(vmi, "#", "network-config", testNetworkData)
+			})
+			It("[test_id:3182]should have cloud-init network-config with NetworkDataBase64 source", func() {
+				vmi := tests.NewRandomVMIWithEphemeralDiskAndNetworkData(
+					cd.ContainerDiskFor(cd.ContainerDiskCirros), testNetworkData, true)
+				LaunchVMI(vmi)
+				tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
 
-				By("checking cloudinit user-data")
-				CheckCloudInitFile(vmi, "#", "user-data", testUserData)
+				By("mouting cloudinit iso")
+				MountCloudInitNoCloud(vmi, "#")
+
+				By("checking cloudinit network-config")
+				CheckCloudInitFile(vmi, "#", "network-config", testNetworkData)
 			})
 			It("[test_id:3183]should have cloud-init network-config from k8s secret", func() {
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdataNetworkData(
-					cd.ContainerDiskFor(cd.ContainerDiskCirros), "", "", false)
+				vmi := tests.NewRandomVMIWithEphemeralDiskAndNetworkData(
```

Comment:
You need to add some network data here

Reply:
That would be silencing the test but not covering the test case. We need to make sure that this works when none of the data is provided directly.

Trying to adjust our tests helpers to that.

Reply:
But you require it here: https://github.com/kubevirt/kubevirt/pull/4034/files#diff-a2a9502e19df0074934cf310404d93b6R397
Am I missing something?

Reply:
Yeah, but should it be required when we provide the data over a Secret?

Reply:
>But you require it here: https://github.com/kubevirt/kubevirt/pull/4034/files#diff-a2a9502e19df0074934cf310404d93b6R397
Am I missing something?

this is technically fine, the secret get's read into the normal data fields during the startup routine.